dashboard/src/components/AppView/SecretsTable/SecretsTable.tsx

```diff
@@ -0,0 +1,74 @@
+import * as _ from "lodash";
+import * as React from "react";
+
+import { ErrorSelector } from "../../../components/ErrorAlert";
+import { IKubeItem, ISecret } from "../../../shared/types";
+import SecretItem from "./SecretItem";
+
+interface IServiceTableProps {
+  namespace: string;
+  secretNames: string[];
+  secrets: { [s: string]: IKubeItem<ISecret> };
+  getSecret: (namespace: string, name: string) => void;
+}
+
+interface IError {
+  resource: string;
+  error: Error;
+}
+
+class SecretsTable extends React.Component<IServiceTableProps> {
+  public async componentDidMount() {
+    const { getSecret, secretNames, namespace } = this.props;
+    secretNames.forEach(s => {
+      getSecret(namespace, s);
+    });
+  }
+
+  public render() {
+    const { secrets } = this.props;
+    const secretKeys = Object.keys(secrets);
+    const secretError = this.findError();
+    return (
+      secretKeys.length > 0 && (
+        <div>
+          <h6>Secrets</h6>
+          <table>
+            <thead>
+              <tr className="flex">
+                <th className="col-2">NAME</th>
+                <th className="col-2">TYPE</th>
+                <th className="col-7">DATA</th>
+              </tr>
+            </thead>
+            <tbody>
+              {secretKeys.map(
+                k => secrets[k].item && <SecretItem key={k} secret={secrets[k].item as ISecret} />,
```

Comment:
This should no longer be needed, right?
```suggestion
                k => secrets[k].item && <SecretItem key={k} secret={secrets[k].item} />,
```

Reply:
It is needed because even if we are checking that `item` is set, TS complain about that it may be undefined:

```
Type 'ISecret | undefined' is not assignable to type 'ISecret'.
  Type 'undefined' is not assignable to type 'ISecret'.
```

Reply:
In that case, we could use the bang operator instead, so `secrets[k].item!`, anyway not a big deal. It's weird that TypeScript isn't able to figure out that it's defined based on the prior condition. Maybe because it's not in a variable?